nvtabular/ops/categorify.py

```diff
@@ -433,23 +433,47 @@ def _get_embedding_order(cat_names):
     return cat_names
 
 
-def get_embedding_sizes(workflow):
-    """Returns a dictionary of best embedding sizes from the workflow"""
+def get_embedding_sizes(source, output_dtypes=None):
+    """Returns a dictionary of embedding sizes from a workflow or column_group
+
+    Parameters
+    ----------
+    source : Workflow or ColumnGroup
+        Either a nvtabular Workflow or ColumnGroup object that we should use to find
+        embedding sizes
+    output_dtypes : dict, optional
+        Optional dictionary of column_name:dtype. If passing a workflow object dtypes
+        will be read from the workflow. This is used to figure out which columns
+        are categorical, which are split out by this function. If passed a column_group
+        and this parameter isn't set, you won't have multihot columns returned separately
+    """
     # TODO: do we need to distinguish multihot columns here?  (if so why? )
-    queue = [workflow.column_group]
+
+    # have to lazy import Workflow to avoid circular import errors
+    from nvtabular.workflow import Workflow
```

Comment:
Can we avoid the circular import by checking `isinstance(source, ColumnGroup)` below?

Reply:
Looks like we'll still have circular import errors =( https://github.com/NVIDIA/NVTabular/blob/main/nvtabular/column_group.py#L20